### DISCUSSION TITLE
Einbindung Tasmota Steckdosen gelingt nicht

### DISCUSSION THREAD

**Discussion by phoenix-83** [camueller/SmartApplianceEnabler#273] [Installation und Betrieb]
Hallo zusammen,

ich bin begeistert von den Möglichkeiten des SAEs und möchte ihn gerne verwenden. 

Aktuell versuche ich zwei Tasmota Steckdosen im SAE anzulegen und im Sunny Portal zu visualisieren. 
Die Einbindung klappt soweit, dass ich die beiden Steckdosen im Portal sehen kann. 
Was bisher noch nicht klappt ist, dass mir keine Verbrauchsdaten angezeigt werden. 

So hab ich das Ganze konfiguriert:

![Tasmota_SAE_01](https://user-images.githubusercontent.com/95980470/170785458-60388c85-96e5-43f4-b869-02951648858a.png)

![Tasmota_SAE_02](https://user-images.githubusercontent.com/95980470/170785462-a6195823-d70d-4edd-b8a2-b3a7f4d3b22e.png)

![Tasmota_SAE_03](https://user-images.githubusercontent.com/95980470/170785465-bd2d0268-02f2-452b-9c14-8c0287e16415.png)

![Tasmota_SunnyPortal_01](https://user-images.githubusercontent.com/95980470/170785467-bca0aab7-4cb8-4d94-a987-2d91d69acc7d.png)

![Tasmota_SunnyPortal_02](https://user-images.githubusercontent.com/95980470/170785469-86ec896a-705c-4325-997e-66da01683d74.png)

Kann mir bitte jemand aufs Pferd helfen, was ich falsch mache? Bin langsam am verzweifeln und hab schon tausend Dinge ausprobiert, die alle nicht zum Erfolg geführt haben.

Danke und Gruß

Heiko

**Comment by Trunks1982** [comment]
So muss das aussehn

![5B8CC093-59D6-4A25-8459-A91C750D7B8B](https://user-images.githubusercontent.com/62701386/170793170-081ba446-ff67-4dd2-8965-6320b2c43879.png)

**Comment by camueller** [comment]
Weil Neu-User des SAE sich häufiiger mit diesem Problem hier im Forum melden habe ich extra eine Seite erstellt, die bei der systematischen Ursachensuche helfen soll: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md

Dein Zähler ist auf `Zählerstand` konfiguriert - das ist ok, aber Du hast zusätzlich zum `Pfad für Extraktion` auch eine `Regex für Extraktion` angegeben - das ist falsch. Letzteres Feld muss leer bleiben. Steht das irgendwo in der Doku so falsch drin??

**Comment by phoenix-83** [comment]
Hallo, 
danke für die schnelle Rückmeldung. 
@camueller nein, das steht nicht falsch in der Doku, ich hatte jedoch schon alles mögliche ausprobiert und hatte diesen Schnipsel mit der Regex für Extraktion irgendwo im photovoltaik forum gelesen. 

Ich hab das jetzt auch mal so eingestellt, wie von @Trunks1982 beschrieben. Leider hat das auch nicht zum Erfolg geführt.

Dann bin ich noch mal die QA durchgegangen. Dort bin ich jetzt soweit, dass ich herausgefunden habe, dass 
- der SAE active running anzeigt laut Status-Abfrage über putty (sudo systemctl status smartapplianceenabler.service)
- der SAE sich pingen lässt
- der SHM sich pingen lässt
- die SEMP-URL den Status = Off zurückgibt:
![SEMP-URL](https://user-images.githubusercontent.com/95980470/170841271-ee180ea5-c459-48e6-a8ae-587992b03e28.png)

Übersehe ich was?

**Comment by camueller** [comment]
Du siehst doch, dass `AveragePower` im SEMP mit 0 angegeben ist. Wenn die Leistung zu dem Zeitpunkt nicht 0 war, musst Du im Log nachschauen, warum das so ist.

**Comment by phoenix-83** [comment]
Kurze Rückmeldung: Es läuft jetzt. 

Das Problem war anscheinend, dass das Paket pigpio nicht installiert war. Wird das im offiziellen SAE Image nicht automatisch mit installiert?

@camueller Danke für deine Geduld :)

**Comment by camueller** [comment]
Du meinst das Docker-Image? Da gibt es tatsächlich noch einen offenen Issue, für den ich bisher noch keine Zeit hatte: https://github.com/camueller/SmartApplianceEnabler/issues/260

**Comment by phoenix-83** [comment]
Nein, ich hab nicht das Docker-Image gemeint, sondern hab das letzte modifizierte Image für den Raspi (2022-01-28-raspios-bullseye-armhf-lite-usbrun.zip) runtergeladen und exakt nach der Anleitung installiert. 

Deswegen war ich verwundert, dass es nicht auf Anhieb funktionierte.

**Comment by camueller** [comment]
Danke für den Hinweis - muss ich mal prüfen ...

**Comment by phoenix-83** [comment]
Ich hab nochmal ne Frage zur Funktionsweise.

Was ich mit der Tasmota Steckdose machen möchte ist, meine Poolpumpe automatisch einschalten, wenn genügend PV-Überschuss vorhanden ist und die Pumpe automatisch abgeschalten, wenn kein Überschuss mehr vorhanden ist. Dann bei genügend Überschuss wieder einschalten usw...

Leider funktioniert das noch nicht wie gewünscht.

Bisher habe ich es soweit, dass die Steckdose im Sunny Portal erkannt und der Verbrauch geloggt wird. 

![Screenshot 2022-06-02 213917](https://user-images.githubusercontent.com/95980470/171724531-d85aae04-2737-4b3c-89f8-a1fc040b8998.png)

Über die Ampel im SAE kann ich den Verbraucher auch manuell starten, jedoch läuft die Pumpe dann solange wie die maximale Laufzeit eingegeben wurde und die PV-Erzeugung wird ignoriert. Sprich sie schaltet nicht ab, obwohl die Pumpe auf Netzbezug läuft.
Was habe ich falsch eingestellt oder nicht verstanden?

Ich habe die Option "Unterbrechung erlaubt" angewählt, was ich so interpretiere, dass der Verbraucher jederzeit ein und ausgeschaltet werden kann, je nach dem ob Überschuss vorhanden ist oder nicht. Oder ist es zwingend notwendig die 4 Zeiten auszufüllen, damit die Funktion geht?

Sorry, aber ich stehe echt auf dem Schlauch?! Werde auch leider aus der Anleitung nicht so recht schlau. Ist es grundsätzlich möglich, einen Verbraucher so wie ich es beschrieben habe zu steuern?

![Screenshot 2022-06-02 213307](https://user-images.githubusercontent.com/95980470/171723599-ebcbe3de-a810-4fef-8683-5342791a3a04.png)

![Screenshot 2022-06-02 213334](https://user-images.githubusercontent.com/95980470/171723609-3fddf04f-b96a-4a53-836a-8f0b6d81d578.png)

**Comment by camueller** [comment]
Grundsätzlich kann man ein Gertä so steuern, wie Du es willst. Deine Konfiguration ist dafür korrekt.
Aber ich glaube nicht, dass Du willst, dass die Pool-Pumpe nicht läuft, wenn es 4 Tage keinen PV-Überschuss gibt. Dann ist Dein Pool grün. Ich habe selbst einen Pool und einfach die gewünschte Zeit eingegeben (8 Stunden täglich), der Home Manager kann dann entscheiden, wann er die Pumpe ein-/ausschaltet.

Hast Du im Sunny Portal eingestellt, dass das Gerät nur mit PV-Strom laufen soll?

![image](https://user-images.githubusercontent.com/4489628/171795163-a5d58e0d-0f8a-43e6-87d6-4a0a533e15b6.png)

**Comment by phoenix-83** [comment]
Da geb ich dir recht, es wäre wirklich nicht gut, wenn die Pumpe über 4 Tage nicht laufen würde...

so sieht meine Konfiguration im Sunny Portal aus:
![grafik](https://user-images.githubusercontent.com/95980470/171801711-8d9d1afd-d89c-464c-a120-6f86cdd1b186.png)

**Comment by camueller** [comment]
Die Filterpumpe für meinen Pool hat diesen Zeitplan:

![image](https://user-images.githubusercontent.com/4489628/171802594-569cd0d7-ee09-4b11-b0ca-eb307a559133.png)

**Comment by phoenix-83** [comment]
Und bei dir läuft die Pumpe vorrangig mit PV Überschuß und nur wenn der nicht reicht auf Netzstrom? oder wie muss ich das verstehen?
Du hast ja keine Mindestlaufzeit eingegeben. Welchen Ampelzustand hat die Pumpe dann morgens bevor genügend PV Überschuss da ist?

**Comment by camueller** [comment]
Lies mal die [Dokumentation](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Schedules_DE.md) aufmerksam: 
Mit meiner Konfiguration läuft die Pumpe jeden Tag 8 Stunden.

**Comment by phoenix-83** [comment]
Das habe ich auch schon alles probiert, das Problem ist nur, dass die Pumpe morgen nicht automatisch startet. Weder mit Netzbezug noch mit PV-Überschuss. Daher bin ich ein wenig ratlos

**Comment by camueller** [comment]
Irgendetwas stimmt mit Deiner Konfiguration nicht, sonst würde es ja funktionieren. Ich habe fast den identischen Anwendungsfall und mein Konfigraiton hatte ich ja oben schon gepostet.

Dann solltest Du mal prüfen, was der SAE an der SEMP-Schnittstelle meldet - schau mal im Q&A-Bereich der SAE-Homepage nach.
